### ISSUE TITLE
Presto Parsing Error

### ISSUE THREAD

**Issue by ruichengnew** [tobymao/sqlglot#1157]
Like #1079, in group clause,  we can use column, rollup, grouping sets, cube, these four types. 

However, when we change the order of the four types, sqlglot cannot parse. 

```python
sqlglot.parse_one(
    """
SELECT origin_state, destination_state, origin_zip, sum(package_weight)
FROM shipping
GROUP BY
    ROLLUP (origin_zip),
    GROUPING SETS ((origin_state, destination_state)), a
""",
    read="presto",
)
```

**Comment by tobymao** [comment]
@ruichengnew you can have the rollup / grouping sets in between the columns of the group by?

**Comment by tobymao** [comment]
fixed by 2d29458
